src/scipp/optimize/__init__.py

```diff
@@ -59,11 +59,47 @@ def _make_defaults(f, p0):
     return kwargs
 
 
+def _get_specific_bounds(bounds, name, unit) -> Tuple[float, float]:
+    if name not in bounds:
+        return -np.inf, np.inf
+    b = bounds[name]
+    if isinstance(b, Variable):
+        if b.shape != [2]:
+            raise ValueError("Parameter bounds must be either a tuple of scalar "
+                             "variables or a 2-element array variable. "
+                             f"Got sizes={b.sizes} as bounds for '{name}'.")
+        return tuple(b.to(unit=unit, dtype=float).values)
```

Comment:
It is a bit odd to support this, I think? What is the meaning of the dimension?

Reply:
None as far as `curve_fit` is concerned. I was just thinking it might be useful, similarly to passing limits as a length-2 array to `sc.array`.
But I don't mind removing it.

Reply:
> similarly to passing limits as a length-2 array to sc.array.

Not sure what you are referring to?

Reply:
I meant `sc.arange`: https://github.com/scipp/scipp/issues/2140#issuecomment-920562796